### PR TITLE
Update uncrustify configuration and improve CI setup (see FreeRTOS/FreeRTOS-Kernel/pull/445)

### PR DESCRIPTION
**Pull request by swaldhoer**
Description
-----------

The uncrustify changes in the Kernel repository have been merged (see FreeRTOS/FreeRTOS-Kernel/pull/445).
This PR just aligns the uncrustify setup of this repository with the one used in the Kernel repository.

The reason for pinning the uncrusitfy version and updating the configuration are the same as in the Kernel repository.

Pinging @alfred2g and @cobusve as you had a look at the PR in Kernel repository.

Test Steps
-----------

Tested is done in CI.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by swaldhoer** [comment]
`#if`s that are followed by a `{`are now rendered the same way as in the Kernel repository.
This is the reason why where are a few files with changes.

It has basically changed from

```c
#if ( BLA == 1 )
    {
        /* code */
    }
#endif /* if ( BLA == 1 ) */
```

to

```c
#if ( BLA == 1 )
{
    /* code */
}
#endif /* if ( BLA == 1 ) */
```

File: FreeRTOS/Demo/Common/Minimal/AbortDelay.c [hunk 1]
```diff
@@ -420,18 +420,18 @@
         EventBits_t xBitsToWaitFor = ( EventBits_t ) 0x01, xReturn;
 
         #if ( configSUPPORT_STATIC_ALLOCATION == 1 )
-            {
-                static StaticEventGroup_t xEventGroupBuffer;
+        {
+            static StaticEventGroup_t xEventGroupBuffer;
```
Comment (alfred2g) [inline FreeRTOS/Demo/Common/Minimal/AbortDelay.c:425]: oh this looks much better now!

**Comment by alfred2g** [comment]
@swaldhoer,  Thanks for the second PR, the code looks much better now!

**Review by alfred2g — APPROVED** [review]

**Comment by alfred2g** [comment]
@swaldhoer, I tool the liberty to modify the license in most of the files as the check was failing.
Thanks

**Comment by swaldhoer** [comment]
@alfred2g should I rebase? Or how do we continue?

**Comment by alfred2g** [comment]
> @alfred2g should I rebase? Or how do we continue?

yeah please!, don't forget to pull the changes from here first.

**Comment by swaldhoer** [comment]
> > @alfred2g should I rebase? Or how do we continue?
> 
> yeah please!, don't forget to pull the changes from here first.

I tried this time to use GitHubs webinterface instead of bash...and then I got a merge main instead of rebase - hope this is also okay 🙄 sorry

**Review by alfred2g — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**